crates/sc-consensus-subspace-rpc/src/lib.rs

```diff
@@ -855,4 +867,63 @@ where
             pipe_from_stream(pending, mapping_stream).boxed(),
         );
     }
+
+    fn subscribe_filtered_object_mappings(
+        &self,
+        pending: PendingSubscriptionSink,
+        hashes: Vec<Blake3HashHex>,
+    ) {
+        if hashes.len() > MAX_OBJECT_HASHES_PER_SUBSCRIPTION {
+            error!(
+                "Request hash count ({}) exceed the server limit: {} ",
+                hashes.len(),
+                MAX_OBJECT_HASHES_PER_SUBSCRIPTION
+            );
+
+            let err_fut = pending.reject(Error::StringError(format!(
+                "Request hash count ({}) exceed the server limit: {} ",
+                hashes.len(),
+                MAX_OBJECT_HASHES_PER_SUBSCRIPTION
+            )));
+
+            self.subscription_executor.spawn(
+                "subspace-filtered-object-mappings-subscription",
+                Some("rpc"),
+                err_fut.boxed(),
+            );
+
+            return;
+        };
+
+        let mut hashes = HashSet::<Blake3Hash>::from_iter(hashes.into_iter().map(|hash| *hash));
+        let hash_count = hashes.len();
+
+        // The genesis segment isn't included in this stream, see
+        // `subscribe_archived_object_mappings` for details.
+        let mapping_stream = self
+            .archived_segment_notification_stream
+            .subscribe()
+            .flat_map(move |archived_segment_notification| {
+                let objects = archived_segment_notification
+                    .archived_segment
+                    .global_object_mappings()
+                    .filter(|object| hashes.remove(&object.hash))
+                    .collect::<Vec<_>>();
+
+                stream::iter(objects)
+            })
+            // Stop when we've returned mappings for all the hashes. Since we only yield each hash
+            // once, we don't need to check if hashes is empty here.
+            .take(hash_count)
+            // Typically batches will be larger than the hash limit, but we want to allow CLI
+            // options to change that.
+            .ready_chunks(OBJECT_MAPPING_BATCH_SIZE)
```

Comment:
This does nothing because it is larger than `MAX_OBJECT_HASHES_PER_SUBSCRIPTION`

Reply:
For now, yes, but later we will make these CLI options and maybe change the defaults. In the meantime it collects into a Vec.